kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/MetricsIT.java

```diff
@@ -152,14 +169,174 @@ void prometheusMetricsWithCommonTags(KafkaCluster cluster) {
 
             var metrics = ahc.scrapeMetrics();
             assertThat(metrics)
-                    .filteredOn("name", counterName)
+                    .filterByName(counterName)
                     .singleElement()
-                    .extracting(SimpleMetric::labels)
-                    .isEqualTo(Map.of("a", "b"));
+                    .labels()
+                    .containsEntry("a", "b");
+        }
+    }
+
+    static Stream<Arguments> transitScenarios() {
+        var decodeAll = new NamedFilterDefinitionBuilder("decodeAll", "DecodeAll").build();
+        return Stream.of(
+                argumentSet("counts opaque requests from client",
```

Comment:
I'm not currently testing that a shorted-circuit request ticks only the downstream end metric.